pkg/resource/sensitive.go

```diff
@@ -198,7 +201,23 @@ func GetSensitiveParameters(ctx context.Context, client SecretClient, from runti
 			}
 
 			switch v.(type) {
-			case []map[string]interface{}, []interface{}:
+			case map[string]interface{}:
+				sel := &map[string]v1.SecretKeySelector{}
+				if err = pavedJSON.GetValueInto(expandedJSONPath, sel); err != nil {
+					return errors.Wrapf(err, errFmtCannotGetSecretKeySelector, expandedJSONPath)
```

Comment:
nit: should we differentiate errors, e.g. `errFmtCannotGetSecretKeySelectorAsMap`, `errFmtCannotGetSecretKeySelectorAsList` ?